### PR TITLE
upgrade elastic 6.7

### LINKED ISSUE #781: update elasticsearch version
to get rid of Hashie Mash warning and keep up with new development

### PR DESCRIPTION
**Pull request by netagonen**
Resolves #781

### CODE REVIEW

**Comment by netagonen** [comment]
@alonpeer , would you like to review?

File: docker-compose.yml [hunk 1]
```diff
@@ -6,10 +6,18 @@ services:
       - postgres_data:/var/lib/postgresql/data
     environment:
       - POSTGRES_PASSWORD=password
+    #when upgrading elasticsearch server on local machine:
+    #network_mode: "host"
```
Comment (alonpeer) [inline docker-compose.yml:10]: These comments can be deleted.

File: docker-compose.yml [hunk 2]
```diff
@@ -6,10 +6,18 @@ services:
...
+      - "5432:5432"
   elastic:
-    image: elasticsearch:2.4.5
-    volumes:
-      - elastic_data:/usr/share/elasticsearch/data
+    image: elasticsearch:6.7.2
+    # we commented out the next 2 lines so version 6 could work
+    # TODO: find out if it's needed / how to make it work
+    # volumes:
+    #   - elastic_data:/usr/share/elasticsearch/data
```
Comment (alonpeer) [inline docker-compose.yml:18]: Is there still anything to check? Or can we delete these comments as well?

File: docker-compose.yml [hunk 3]
```diff
@@ -21,6 +29,8 @@ services:
     volumes:
       - .:/speakerinnen_liste
       - bundle:/bundle
+    #when upgrading elasticsearch server on local machine:
+    #network_mode: "host"
```
Comment (alonpeer) [inline docker-compose.yml:33]: These can also be deleted.

File: docker-compose.yml [hunk 4]
```diff
@@ -31,4 +41,4 @@ services:
 volumes:
   bundle:
   postgres_data:
-  elastic_data:
+  # elastic_data:
```
Comment (alonpeer) [inline docker-compose.yml:44]: 🧹

File: app/models/concerns/searchable.rb [hunk 5]
```diff
@@ -85,16 +84,19 @@ def self.search(query, filter_countries, filter_cities, filter_lang)
...
         }
 
-      # minimum score depends completely on the given data and query, find out what works in your case.
+      # minimum score:
+      # depends completely on the given data and query, find out what works in your case.
       # can't be integrated directly in query hash because tests fail
-      query_hash[:min_score] = 0.08 if Rails.env.production?
+      # query_hash[:min_score] = 0.08 if Rails.env.production?
+      # version migration: scoring changed from 2.4.5 to 6.0, so 0.08 won't filter.
+      # if min_score stays make it higher (i.e. 3.00)
```
Comment (alonpeer) [inline app/models/concerns/searchable.rb:99]: I think it's good to have this comment as a PR comment, and then remove these three lines from the code. Going forward, we won't need to know the history of this in the code, and can always look at the PR for more context of the changes made.

File: app/models/concerns/searchable.rb [hunk 6]
```diff
@@ -85,16 +84,19 @@ def self.search(query, filter_countries, filter_cities, filter_lang)
...
+                field: 'country.keyword',
                 size: 999
               }
             }
           }
         }
 
-      # minimum score depends completely on the given data and query, find out what works in your case.
+      # minimum score:
+      # depends completely on the given data and query, find out what works in your case.
```
Comment (alonpeer) [inline app/models/concerns/searchable.rb:95]: These lines can also be deleted, if we're removing the `min_score`.

**Review by alonpeer — APPROVED** [review]
Looks good! Just a few minor comments.

A small suggestion for performance improvement: in `searchable.rb` we have `explain: true`, which means that each time a search page is loaded, the query to elasticsearch returns not just the results, but also a big chuck of JSON with explanation of the results.
This slows down the page load for 2 main reasons:
1. The size of the response that elasticsearch sends back is larger, which means it takes longer to be sent (network I/O).
2. Ruby has to parse a larger string into a JSON object, which also takes longer (CPU).

I'm assuming that we're not constantly looking at the app's production logs for these explanations for each of our visitors' queries, but this may be interesting for the development environment, or for when we log in to Heroku and wish to execute these searches from the command line.

So I suggest the following changes:

1. Change the `self.search` method signature to be `def self.search(query, filter_countries, filter_cities, filter_lang, with_explain)`, and change the `query_hash` param to be `explain: with_explain`.
2. Change the call to this method in `profiles_controller.rb` to be `.search(params[:search], params[:filter_countries], params[:filter_cities], params[:filter_lang], !Rails.env.production? || params[:explain])`.

This means that for normal searches in production, `explain` will be `false`, unless one adds `&explain=1` to the URL, and for any other environments, e.g. development, `explain` will be `true`.

File: docker-compose.yml (hunk 2 above)
Comment (netagonen) [reply docker-compose.yml:18]: we still need to check. the volumes are not a must but a nice to have, so one won't have to recreate the index every time they delete the container

File: docker-compose.yml (hunk 4 above)
Comment (netagonen) [reply docker-compose.yml:44]: this might be needed for the volumes

**Review by alonpeer — APPROVED** [review]

**Review by zaziemo — APPROVED** [review]
🎉 🎉 👨🏽‍💻👩🏽‍💻 Thanks for your work @netagonen & @alonpeer!

**Comment by alonpeer** [comment]
Congrats on deploying this!!!

GIF

**Pull request merged**